osu.Game/Skinning/SkinManager.cs

```diff
@@ -289,46 +283,23 @@ public Task<ILive<SkinInfo>> Import(SkinInfo item, ArchiveReader archive = null,
 
         #region Implementation of IModelManager<SkinInfo>
 
-        public event Action<SkinInfo> ItemUpdated
-        {
-            add => skinModelManager.ItemUpdated += value;
-            remove => skinModelManager.ItemUpdated -= value;
-        }
-
-        public event Action<SkinInfo> ItemRemoved
-        {
-            add => skinModelManager.ItemRemoved += value;
-            remove => skinModelManager.ItemRemoved -= value;
-        }
-
-        public void Update(SkinInfo item)
-        {
-            skinModelManager.Update(item);
-        }
-
-        public bool Delete(SkinInfo item)
+        public void Delete([CanBeNull] Expression<Func<SkinInfo, bool>> filter = null, bool silent = false)
         {
-            return skinModelManager.Delete(item);
-        }
-
-        public void Delete(List<SkinInfo> items, bool silent = false)
-        {
-            skinModelManager.Delete(items, silent);
-        }
+            using (var context = contextFactory.CreateContext())
+            {
+                var items = context.All<SkinInfo>()
+                                   .Where(s => !s.Protected && !s.DeletePending);
+                if (filter != null)
+                    items = items.Where(filter);
 
-        public void Undelete(List<SkinInfo> items, bool silent = false)
-        {
-            skinModelManager.Undelete(items, silent);
-        }
+                // check the removed skin is not the current user choice. if it is, switch back to default.
+                Guid currentUserSkin = CurrentSkinInfo.Value.ID;
 
-        public void Undelete(SkinInfo item)
-        {
-            skinModelManager.Undelete(item);
-        }
+                if (items.Any(s => s.ID == currentUserSkin))
+                    scheduler.Add(() => CurrentSkinInfo.Value = Skinning.DefaultSkin.CreateInfo().ToLive());
```

Comment:
What's this schedule about? I would ideally not want to see it there at all but guessing it may not be so simple.

Reply:
That's a game wide bindable used to track the skin, which was previously relied on being update thread set only.

I can go through all cases and schedule usages, but maybe in a separate cleanup PR? I don't think that bindable should live in `SkinManager` in the first place (similar to the global beatmap).

Reply:
Fair, I agree with not changing that here.